GEECS-PythonAPI/geecs_python_api/controls/data_acquisition/multiscan_analysis.py

```diff
@@ -0,0 +1,294 @@
+# %% imports
+import os
+import glob
+import cv2
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+from geecs_python_api.controls.data_acquisition.data_acquisition import DataInterface
+from geecs_python_api.controls.data_acquisition.scan_analysis import VisaEBeamAnalysis
+from image_analysis.utils import read_imaq_image
+
+# =============================================================================
+# %% classes
+
+class MultiScanAnalysis():
+    """
+    Intended to be a parent class for generalized multi-scan analysis.
+    Not sure how to generalize at this point.
+    """
+
+    def __init__(self):
+        self.data_interface = DataInterface()
+
+    def parse_scans(self, input_scans):
+        """
+        Parse input scans and organize.
+        """
+
+        # parse input
+        scan_numbers = list(input_scans['scan_num'])
+        scan_years = ([input_scans['year']] * len(scan_numbers)
+                      if isinstance(input_scans['year'], str)
+                      else list(input_scans['year']))
+        scan_months = ([input_scans['month']] * len(scan_numbers)
+                      if isinstance(input_scans['month'], str)
+                      else list(input_scans['month']))
+        scan_days = ([input_scans['day']] * len(scan_numbers)
+                      if isinstance(input_scans['day'], str)
+                      else list(input_scans['day']))
+
+        # check compatibility of scan dates
+        if (len(scan_years) != len(scan_numbers)
+            or len(scan_months) != len(scan_numbers)
+            or len(scan_days) != len(scan_numbers)):
+            raise Exception("Error: Incompatible dates specifications.")
+
+        # organize isolated scan dicts
+        scan_dict = {}
+        for ind, scan in enumerate(scan_numbers):
+            scan_dict[scan] = {'day': scan_days[ind],
+                               'month': scan_months[ind],
+                               'year': scan_years[ind],
+                               'num': scan}
+
+            (scan_dict[scan]['path_scan'],
+             scan_dict[scan]['path_analysis'],
+             scan_dict[scan]['path_sfile']) = self.get_scan_paths(scan_dict[scan])
+
+        return scan_dict
+
+    def get_scan_paths(self, scan_info):
+
+        self.data_interface.year = scan_info['year']
+        self.data_interface.month = scan_info['month']
+        self.data_interface.day = scan_info['day']
+
+        raw_data_path, analysis_path = self.data_interface.create_data_path(scan_info['num'])
+        scan_directory = raw_data_path / f"Scan{scan_info['num']:03d}"
+        analysis_directory = analysis_path / f"Scan{scan_info['num']:03d}"
+        sfile_path = analysis_path / f"s{scan_info['num']}.txt"
+
+        return scan_directory, analysis_directory, sfile_path
+
+    @staticmethod
+    def load_sfile(filepath):
+        dataframe = pd.read_table(filepath, header=0)
+        return dataframe
+
```

Comment:
More down-the-road when we generalize things, it would be good to define classes that every multiscan analysis should have implemented.   In that train of thought, a blank `analyze()` function that just returns an error when it is called could be good, and then the idea would be that all child classes would need to write their own `analyze()` function to avoid that error.

Reply:
or `run_routine()` as you have at the bottom of this script